content/300-guides/050-database/870-using-prisma-with-mongodb.mdx

```diff
@@ -0,0 +1,326 @@
+tocDepth: 2
+toc: true
+---
+
+<TopBlock>
+
+This guide discusses the concepts behind using Prisma and MongoDB, explains the commonalities and differences between MongoDB and other database providers, and leads you through the process for configuring your application to integrate with MongoDB using Prisma.
+
+</TopBlock>
+
+## What is MongoDB?
+
+MongoDB is a NoSQL database that stores data in [BSON](https://bsonspec.org/) format, a JSON-like document format designed for storing data in key-value pairs. It is commonly used in JavaScript application development because the document model maps easily to objects in application code, and there is built in support for high availability and horizontal scaling.
+
+MongoDB stores data in collections that do not need a schema to be defined in advance, as you would need to do with tables in a relational database. The structure of each collection can also be changed over time. This flexibility can allow rapid iteration of your data model, but it does mean that there are a number of differences when using Prisma to work with your MongoDB database.
+
+## Commonalities with other database providers
+
+Some aspects of using Prisma with MongoDB are the same as when using Prisma with a relational database. You can still:
+
+- model your database with the [Prisma Schema Language](/concepts/components/prisma-schema)
+- connect to your database, using the [`mongodb` database connector](/concepts/database-connectors/mongodb)
+- use [Introspection](/concepts/components/introspection) for existing projects if you already have a MongoDB database
+- use [`db push`](/concepts/components/prisma-migrate/db-push) to push changes in your schema to the database
+- use [Prisma Client](/concepts/components/prisma-client) in your application to query your database in a type safe way based on your Prisma Schema
+
+## Differences to consider
+
+MongoDB's document-based structure and flexible schemas means that using Prisma with MongoDB differs from using it with a relational database in a number of ways. These are some areas where there are differences that you need to be aware of:
+
+- **Defining IDs**: MongoDB documents have an `_id` field (that often contain an [ObjectID](https://www.mongodb.com/docs/manual/reference/bson-types/#std-label-objectid)). This needs to be mapped to a Prisma field. For more information, see [Defining IDs in MongoDB](/concepts/components/prisma-schema/data-model#defining-ids-in-mongodb).
+
+- **Migrating existing data to match your Prisma schema**: In relational databases, all your data must match your schema. If you change the type of a particular field in your schema when you migrate, all the data must also be updated to match. In contrast, MongoDB does not enforce any particular schema, so you need to take care when migrating. For more information, see [How to migrate old data to new schemas](#how-to-migrate-existing-data-to-match-your-prisma-schema).
```

Comment:
> all the data must also be updated to match

Usually the database takes care of that automatically, by casting the data. (Correct @tomhoule?)

Reply:
I don't think it's general enough to be worth mentioning, casting is very limited and there is no casting for objects of the wrong shape, missing fields, etc.